crates/ra_hir_def/src/nameres/tests/globs.rs

```diff
@@ -229,3 +229,149 @@ fn glob_enum_group() {
     "###
     );
 }
+
+#[test]
+fn glob_shadowed_def() {
+    let db = TestDB::with_files(
+        r###"
+        //- /lib.rs
+        mod foo;
+        mod bar;
+
+        use foo::*;
+        use bar::Baz;
+
+        //- /foo.rs
+        pub struct Baz;
+
+        //- /bar.rs
+        pub struct Baz;
+        "###,
+    );
+    let krate = db.test_crate();
+
+    let crate_def_map = db.crate_def_map(krate);
+    let (_, root_module) = crate_def_map
+        .modules
+        .iter()
+        .find(|(_, module_data)| module_data.parent.is_none())
+        .expect("Root module not found");
+    let visible_entries = root_module.scope.entries().collect::<Vec<_>>();
+    insta::assert_debug_snapshot!(
+        visible_entries,
+        @r###"
+    [
+        (
+            Name(
+                Text(
+                    "Baz",
+                ),
+            ),
+            PerNs {
+                types: Some(
+                    (
+                        AdtId(
+                            StructId(
+                                StructId(
+                                    1,
```

Comment:
I wasn't sure if there was an easier way to do this. The important value is the `1` here. Before this change, it would be a `0`, which implies that the name was previously incorrectly resolved from the glob import.

Reply:
I think a way to test thart via usual `def_map` is to add ambiguity between module and type `mod foo;`, `struct foo;` and then checking that `foo::SomeThing` resolves (ie, that we pick module)

Reply:
That worked! Before this change the `Bar` line resolves to `Bar: _`.